server/xpub-controller/helpers/files.test.js

```diff
@@ -1,25 +1,70 @@
+const config = require('config')
+const uuid = require('uuid')
+const { createTables } = require('@pubsweet/db-manager')
 const FilesHelper = require('./files')
+const ManuscriptModel = require('@elifesciences/xpub-model').Manuscript
+
+const filesHelper = new FilesHelper(config)
+
+describe('FilesHelper', () => {
+  describe('validateFileSize', () => {
+    const maxSize = config.get('fileUpload.maxSizeMB')
+    const invalidFileSize = (maxSize + 1) * 1e6
+    const validFileSize = (maxSize - 1) * 1e6
+
+    it('throws an error if the file size is larger than maximum size', () => {
+      expect(() => filesHelper.validateFileSize(invalidFileSize)).toThrow(
+        `File size shouldn't exceed ${maxSize}MB`,
+      )
+    })
+    it('does not throw if the file size is less than MaxSize', () => {
+      expect(() => filesHelper.validateFileSize(validFileSize)).not.toThrow()
+    })
+  })
+
+  describe('generateFileEntity', () => {
+    it('returns the file stream and file entity', async () => {
+      const userId = uuid()
+      await createTables(true)
+      const { id } = await new ManuscriptModel({ createdBy: userId }).save()
+      const file = {
+        stream: {},
+        filename: 'filename',
+        mimetype: 'mimetype',
+      }
+      const filePromise = Promise.resolve(file)
+
+      const { stream } = file
+      const fileData = await FilesHelper.generateFileEntity(filePromise, id)
+      expect(fileData.stream).toEqual(stream)
+      expect(fileData.fileEntity).toMatchObject({
+        manuscriptId: id,
+        id: expect.any(String),
```

Comment:
We could check the length here?

Reply:
I agree, but I think even checking the length is not enough, we should be able to check if this is the exact id we've created, so this is going to change as soon as we start jest mocking instead of using the actual models. I will change it to check the length for now.